kaggle_environments/envs/halite/test_halite.py

```diff
@@ -25,9 +26,121 @@ def test_halite_completes():
+    assert json["statuses"] == ["DONE", "DONE"]
+
+
+def create_board(size=3, starting_halite=0, agent_count=2):
+    env = make("halite", configuration={"size": size, "startingHalite": starting_halite})
+    return Board(env.reset(agent_count)[0].observation, env.configuration)
+
+
+def test_move_moves_ship():
+    size = 3
+    board = create_board(size, agent_count=1)
+    for ship in board.current_player.ships:
+        ship.next_action = ShipAction.SOUTH
+    next_board = board.next()
+    for ship in board.ships.values():
+        next_position = wrap_point(translate_point(ship.position, ShipAction.SOUTH.to_point()), size)
+        next_ship = next_board.ships[ship.id]
+        assert next_ship.position == next_position
+
+
+def move_toward(ship, target: Point):
+    (x1, y1) = ship.position
+    (x2, y2) = target
+    if x2 > x1:
+        return ShipAction.EAST
+    elif x2 < x1:
+        return ShipAction.WEST
+    elif y2 > y1:
+        return ShipAction.NORTH
+    elif y2 < y1:
+        return ShipAction.SOUTH
+
+
+def test_equal_ship_collision_destroys_both_ships():
+    size = 3
+    board = create_board(size, agent_count=2)
+    for ship in board.ships.values():
+        ship.next_action = move_toward(ship, (1, 1))
+    next_board = board.next()
+    assert len(next_board.ships) == 0
+
+
+def test_unequal_ship_collision_destroys_weaker_ship():
+    board = create_board(agent_count=2)
+    for opponent in board.opponents:
+        for ship in opponent.ships:
+            # Make the opponents' ships have more halite so they'll be destroyed
+            ship._halite = 1000
+    for ship in board.ships.values():
+        ship.next_action = move_toward(ship, (1, 1))
+    next_board = board.next()
+    assert len(next_board.current_player.ships) == 1
+    assert len(next_board.ships) == 1
+
+
+def first(iterable):
+    return next(iter(iterable))
+
+
+def test_ship_shipyard_collision_destroys_both():
+    board = create_board(agent_count=2)
+    player_ship = first(board.current_player.ships)
+    opponent_ship = first(first(board.opponents).ships)
+    opponent_ship.next_action = ShipAction.CONVERT
+    board = board.next()
+    assert len(board.ships) == 1
+    assert len(board.shipyards) == 1
+    while player_ship.id in board.ships:
+        board.ships[player_ship.id].next_action = move_toward(player_ship, opponent_ship.position)
+        board = board.next()
+    assert len(board.ships) == 0
+    assert len(board.shipyards) == 0
+
+
+def test_cells_regen_halite():
+    board = create_board(starting_halite=1000, agent_count=1)
+    cell = first(board.cells.values())
+    next_board = board.next()
+    next_cell = next_board[cell.position]
+    expected_regen = round(cell.halite * board.configuration.regen_rate, 3)
+    # We compare to a floating point value here to handle float rounding errors
+    assert next_cell.halite - cell.halite - expected_regen < .000001
+
+
+def test_no_move_on_halite_gathers_halite():
```

Comment:
There are lots of other scenarios we could test in here (like multiship collisions on shipyards, spawning collisions, etc) but I tried to cover most core rules with this pass.